helperFunctions/subscribeHelpers/SalesforceAgentAssignedHelpers/SubscribeToLiveAgentHelper.ts

```diff
@@ -69,24 +75,50 @@ export class SubscribeToLiveAgent {
 						if (chatEndedReason === 'agent') {
 							await updateRoomCustomFields(this.data.room.id, { agentEndedChat: true }, this.read, this.modify);
 							this.modify.getAnalytics().sendEvent(getEventData(this.data.room.id, EventName.CHAT_CLOSED_BY_AGENT));
+							await handleEndChatCallback.handleEndChat(CloseMessages.CLOSED_BY_AGENT);
+						} else {
+							await handleEndChatCallback.handleEndChat(CloseMessages.CLOSED_BY_VISITOR);
+						}
+					} else if (isChatRequestFail === true) {
+						const {
+							message: { reason: chatEndedReason },
+						} = getForEvent(messageArray, 'ChatRequestFail');
+						if (chatEndedReason === 'Unavailable') {
+							await updateRoomCustomFields(this.data.room.id, { agentUnavailable: true }, this.read, this.modify);
+							this.modify
+								.getAnalytics()
+								.sendEvent(getEventData(this.data.room.id, EventName.ESCALATION_FAILED_DUE_TO_NO_LIVEAGENT_AVAILABLE));
+							const noLiveagentAvailableMessage: string = await getAppSettingValue(
+								this.read,
+								AppSettingId.NO_LIVEAGENT_AGENT_AVAILABLE_MESSAGE,
+							);
+
+							await handleEndChatCallback.handleEndChat(noLiveagentAvailableMessage);
+						} else {
+							await handleEndChatCallback.handleEndChat(CloseMessages.UNKNOWN_ERROR_IN_CHECKING_AGENT_RESPONSE);
 						}
-						await handleEndChatCallback.handleEndChat();
 					} else {
 						await messageFilter(this.modify, this.read, this.persistence, this.data.room, this.data.agent, this.assoc, messageArray);
 						const { persistentAffinity, persistentKey } = await retrievePersistentTokens(this.read, this.assoc);
 						if (persistentAffinity !== null && persistentKey !== null) {
 							await this.subscribeToLiveAgent();
 						} else {
 							console.log(ErrorLogs.LIVEAGENT_SESSION_EXPIRED);
-							await handleEndChatCallback.handleEndChat();
+							this.modify.getAnalytics().sendEvent(getEventData(this.data.room.id, EventName.CHAT_CLOSED_LIVEAGENT_SESSION_EXPIRED));
+							await handleEndChatCallback.handleEndChat(CloseMessages.LIVEAGENT_SESSION_EXPIRED);
 							return;
 						}
 					}
 				}
 			})
 			.catch(async (error) => {
-				console.error(ErrorLogs.UNKNOWN_ERROR_IN_CHECKING_AGENT_RESPONSE, error);
-				await handleEndChatCallback.handleEndChat();
+				const event = getEventData(this.data.room.id, EventName.FAILURE_GETTING_SF_MESSAGES);
+				if (event?.properties) {
+					event.properties.reason = error;
+				}
+				this.modify.getAnalytics().sendEvent(event);
+				console.error(ErrorLogs.FAILURE_GETTING_SF_MESSAGES, error);
```

Comment:
Here `event.properties.reason = error;` reason should be some static value, right? something like `failure getting SF messages`

If we pass an error, it can contain roomId which changes every time. So it will not appear in grafana correctly. It will just create thousands of different reasons why the session closed.(Imagine the pie chart it will create 😅 ) This event will not be that frequent but still.